### PR TITLE
Ionic support + API updates

### PR DESCRIPTION
**Pull request by vvoicehovics**
### Description of Changes

1. Remove irrelevant manifest modifications (apparently inherited from the early sdk versions)
2. minor tweaks to types in the ExampleApp. ExampleApp works with cordova, not ionic
3. add signOutUser and pushUnregister API. Add to ExampleApp
4. update android base sdk `6.* --> 7.0.1`, use new api for requesting push token (pushRegister --> pushRequestDeviceToken)
5. when presenting in-apps return OptimoveInAppPresentationResult similar to other sdks

### Breaking Changes

- pushRegister --> pushRequestDeviceToken
- presenting an in-app message always returns a resolved promise with OptimoveInAppPresentationResult 

### Release Checklist

Prepare:

-   [x] `cd cordova-sdk` and `npm run build` to produce minified artifacts
-   [x] `cd ExampleApp` and `npm run build` to update `index.js`
-   [x] Detail any breaking changes. Breaking changes require a new major version number
-   [x] Update type declarations in `cordova-sdk/index.d.ts`

Bump versions in:

-   [x] package.json
-   [x] plugin.xml
-   [x] src/ios/OptimoveSDKPlugin.swift
-   [x] src/android/OptimoveInitProvider.java

Release:

-   [ ] Squash and merge to master
-   [ ] Delete branch once merged
-   [ ] Create tag from master matching chosen version
-   [ ] Fill out release notes
-   [ ] Run `npm publish --access public`

Update wiki:

- [ ] https://github.com/optimove-tech/Optimove-SDK-Cordova/wiki

### CODE REVIEW

File: cordova-sdk/package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     ]
   },
   "devDependencies": {
+    "@optimove-inc/cordova-sdk": "file:../cordova-sdk",
```
Comment (cgwyllie) [inline cordova-sdk/package.json:28]: What's the meaning to depend on the SDK in the SDK? (would expect in the sample app, but here seems strange, maybe it's correct, just don't understand)

File: cordova-sdk/plugin.xml [hunk 2]
```diff
@@ -53,16 +53,7 @@
             <!-- Optimove initProvider-->
             <provider android:authorities="${applicationId}.optimoveinitprovider" android:name="com.optimove.android.cordova.OptimoveInitProvider" android:exported="false" />
         </config-file>
-        <edit-config file="AndroidManifest.xml" target="/manifest" mode="merge">
-            <manifest xmlns:tools="http://schemas.android.com/tools" />
```
Comment (cgwyllie) [inline cordova-sdk/plugin.xml:57]: We don't use `tools:` nodes for anything else? (I think no, just confirming)

**Review by cgwyllie — APPROVED** [review]
LGTM

File: cordova-sdk/package.json (hunk 1 above)
Comment (vvoicehovics) [reply cordova-sdk/package.json]: ofc it's not correct, removed 😬

File: cordova-sdk/plugin.xml (hunk 2 above)
Comment (vvoicehovics) [reply cordova-sdk/plugin.xml:57]: let's start by taking down edit-config removing backup rules

**Pull request merged**